File: app/src/main/java/com/shui520it/access/funcion/action/ShuiRectActionImpl.java
package com.shui520it.access.funcion.action;

import android.graphics.Rect;
import android.view.accessibility.AccessibilityNodeInfo;

import androidx.annotation.NonNull;

/**
 * @author shuimu{lwp}
 * @time 2019/8/14  11:59
 * @desc 获取指定区域
 */
public class ShuiRectActionImpl extends ShuiBaseAction<Rect> {
    private Rect compareData;
    @Override
    public boolean checkOk(AccessibilityNodeInfo thisInfo) {
        if (compareData != null) {
            thisInfo.getBoundsInScreen(compareData);
            return acitonKey.contains(compareData);
        }
        return false;
    }
}

File: app/src/main/java/com/shui520it/access/funcion/action/ShuiRectActionImpl.kt
package com.shui520it.access.funcion.action

import android.graphics.Rect
import android.view.accessibility.AccessibilityNodeInfo

/**
 * @author shuimu{lwp}
 * @time  2021/2/1 14:10
 * @desc
 */
class ShuiRectActionImpl : ShuiBaseAction<Rect>() {
    private var compareRect: Rect? = null
    override fun checkOk(info: AccessibilityNodeInfo): Boolean {
        val result = compareRect?.let {
            info.getBoundsInScreen(it)
            actionKey?.contains(it)
        }
        return result ?: false
    }
}